File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_100549/python/001_csv_to_json/solution.py
import csv
import json
import sys

def csv_to_json():
    input_file = 'input/data.csv'
    data_list = []

    try:
        with open(input_file, mode='r', encoding='utf-8') as f:
            reader = csv.DictReader(f)
            for row in reader:
                # Transform values to specified types
                transformed_row = {
                    "Name": row["Name"],
                    "Age": int(row["Age"]),
                    "Email": row["Email"],
                    "Score": float(row["Score"])
                }
                data_list.append(transformed_row)
        
        # Output valid JSON to stdout
        sys.stdout.write(json.dumps(data_list))
    except Exception:
        # Silent fail as per the constraint "Do not output anything else" 
        # beyond valid JSON, although in a real scenario we'd handle errors.
        pass

if __name__ == "__main__":
    csv_to_json()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b-mlx_think/20260910_100549/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * This program reads a CSV file from 'input/data.csv', 
 * parses its contents based on the specified schema, 
 * and outputs a JSON array to stdout.
 */

function main() {
    const filePath = path.join('input', 'data.csv');

    try {
        // Read the content of the CSV file
        const data = fs.readFileSync(filePath, 'utf8');
        
        // Split into lines and filter out any trailing empty lines
        const lines = data.split(/\r?\n/).filter(line => line.trim() !== '');
        
        if (lines.length === 0) {
            console.log(JSON.stringify([]));
            return;
        }

        // The first line is the header, which we skip
        const rows = lines.slice(1);
        
        const result = rows.map(row => {
            const columns = row.split(',');
            
            // Extract fields based on the expected positions:
            // Name: index 0, Age: index 1, Email: index 2, Score: index 3
            return {
                Name: columns[0]?.trim() || "",
                Age: parseInt(columns[1]?.trim() || "0", 10),
                Email: columns[2]?.trim() || "",
                Score: parseFloat(columns[3]?.trim() || "0")
            };
        });

        // Output the final array as valid JSON
        process.stdout.write(JSON.stringify(result, null, 2) + '\n');
    } catch (err) {
        // If there's an error reading the file, we exit quietly or handle it
        // as per requirements (output must be valid JSON or nothing else).
        process.exit(1);
    }
}

main();